crates/celo-revm/src/common/fee_currency_context.rs

```diff
@@ -1,25 +1,56 @@
+use crate::{
+    CeloContext, CeloEvm,
+    core_contracts::{CoreContractError, get_currencies, get_exchange_rates, get_intrinsic_gas},
+};
 use alloy_primitives::map::HashMap;
-use revm::primitives::{Address, U256};
+use revm::{
+    Database, Inspector,
+    context_interface::ContextTr,
+    handler::EvmTr,
+    primitives::{Address, U256},
+};
 
 use std::{format, string::String};
 
 #[derive(Debug, Clone, Default)]
 pub struct FeeCurrencyContext {
     exchange_rates: HashMap<Address, (U256, U256)>,
     intrinsic_gas: HashMap<Address, U256>,
+    pub updated_at_block: Option<u64>,
```

Comment:
An alternative would be to make the whole `FeeCurrencyContext` optional in the CeloBlockEnv. The would be more accurate but might require more special cases in the future. This way it defaults to "not initialized" == "no supported fee currencies", which seems reasonable.